book/src/libs/wasp/benchspy/simplest_metrics.md

```diff
@@ -0,0 +1,59 @@
+# BenchSpy - Simplest Metrics
+
+As mentioned earlier, `BenchSpy` doesn't include any built-in comparison logic. It's up to you to decide how to compare metrics, as there are various ways to approach it and different data formats returned by queries.
+
+For example, if your query returns a time series, you could:
+- Compare each data point in the time series individually.
+- Compare aggregates like averages, medians, or min/max values of the time series.
+
+Each of these approaches has its pros and cons, and `BenchSpy` doesn't make any judgments here. In this example, we'll use a very simplified approach, which **should not be treated** as a gold standard. In our case, the `QueryExecutor` returns a single data point for each metric, eliminating the complexity. However, with `Loki` and `Prometheus`, things can get more complicated.
+
+## Working with Built-in `QueryExecutors`
+
+Since each built-in `QueryExecutor` returns a different data type, and we use the `interface{}` type to reflect this, convenience functions help cast these results into more usable types:
+
+```go
+currentAsFloat64 := benchspy.MustAllDirectResults(currentReport)
+previousAsFloat64 := benchspy.MustAllDirectResults(previousReport)
+```
+
+> [!NOTE]
+> All standard metrics for the `DirectQueryExecutor` have the `float64` type.
+
+## Defining a Comparison Function
+
+Next, let's define a simple function to compare two floats and ensure the difference between them is smaller than 1%:
+
+```go
+var compareValues = func(
```

Comment:
Can we have a default function for the common case and compare results of generators to previous results of generators or should we write it every time because there are no common case?

Reply:
I'd like just to set thresholds, like no more than 10% latency deviation for 50th, 15% for 95th, and always fail if the errors threshold is reached. Does it make sense, just easier API